prime-angular-frontend/src/app/modules/site-registration/pages/organization-agreement/organization-agreement.component.ts

```diff
@@ -141,6 +141,9 @@ export class OrganizationAgreementComponent implements OnInit, IPage {
     const organization = this.organizationService.organization;
     const siteId = this.route.snapshot.params.sid;
     this.isCompleted = organization?.completed;
+    if (!organization) {
```

Comment:
@johnnyl6  my only feelings on the null checks like this are if the main modal (site, org, enrolment, etc) doesn't exist the page should fail since you shouldn't be able to make it to the page without that model.  So if we go this route we'll to update the templates to be user friend when this occurs by hiding content and showing toasts, or we don't worry about null checks on the main models and add the mocks to each test, as well as, the schema for shallow testing.

Reply:
Agreed. The component is only shared within PRIME and unlikely to be reused as a truly 3rd party lib, which requires much more validations on external input values. Will be going to the route to modify the test with valid mocks.